File: 02.CSharpPartTwo/06_StringsAndTextProcessing/StringsAndTextProcessing/02ReverseString/ReverseString.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/*Problem 2. Reverse string

Write a program that reads a string, reverses it and prints the result at the console.
 */

namespace _02ReverseString
{
    class ReverseString
    {
        static void Main(string[] args)
        {
            Console.Write("Enter the string: ");
            string inputString = Console.ReadLine();

            var stringArray = inputString.ToArray();

            Array.Reverse(stringArray);

            var result = string.Join("", stringArray);

            Console.WriteLine(result);
        }
    }
}

File: 06_JavaScript/09_Strings/Strings/01.ReverseString/ReverseString.js
var inputString = 'sample';

function reverseString(input) {
    var tempString = '',
        i,
        j,
        len = input.length;

    for (i = 0, j = len - 1; i < len; i += 1, j -= 1) {
        tempString += input[j];
    }

    return tempString;
}

console.log(reverseString(inputString));